### ISSUE TITLE
ResNet50: difference between PyTorch and default Metalhead implementation

### ISSUE THREAD

**Issue by Alexander-Barth** [FluxML/Metalhead.jl#110]
I noticed some differences between the PyTorch and the default Metalhead implementation for ResNet50.

* The `connection`defaults to  `connection = (x, y) -> @. relu(x) + relu(y)` while in PyTorch we have essentially `connection = (x, y) -> @. relu(x + y)` Unfortunately, the `ResNet50` constructor does not give access to the `connection` parameter.

https://github.com/pytorch/vision/blob/6db1569c89094cf23f3bc41f79275c45e9fcb3f3/torchvision/models/resnet.py#L119

* If we follow Pytorch, then the first `Conv((7,7))` layer should have no bias I believe:
https://github.com/pytorch/vision/blob/6db1569c89094cf23f3bc41f79275c45e9fcb3f3/torchvision/models/resnet.py#L146

Are there any particular reasons for this difference?

If we would follow the same structure than PyTorch, then we could (more easily) use the same weight as the PyTorch pretrained model. I have made a test with the PyTorch's ResNet50 and a custom Flux' ResNet50 model by reading all the parameters using the PyTorch model `state_dict()` method and this approach works quite well and I get exactly the same results from PyTorch than from Flux.

**Comment by Alexander-Barth** [comment]
I have tested it on ResNet50, 101 and 152. Essentially it goes thought the Flux model and generate a similar data structure and creates an `OrderedDict` similar to PyTorch `model.state_dict()`. It could support other models which are build out of `Chain`, `Parallel`, `BatchNorm`  and `Dense` blocks (and other layers without parameters like pooling layers). But adding other types of layers should not be too difficult as one essentially needs to define in which order the PyTorch serializes the its parameter.

I put my script here:
https://gist.github.com/Alexander-Barth/3f17e6f59daebe94f548833fa3d01f0e#file-test_resnet-jl

I can also make a PR (maybe to [MetalheadWeight](https://github.com/FluxML/MetalheadWeights)) ...

To add other type of layers one would need to add an additional function like this here:
https://gist.github.com/Alexander-Barth/3f17e6f59daebe94f548833fa3d01f0e#file-test_resnet-jl-L113


Of course, it all supposes that the order of parameters is the same (in particular that for example the 1st branch of a `Parallel` layers is also first defined in PyTorch) , but this seems to be the case for the ResNet model I tried. If this is not the case, then the shapes of the parameters will likely not match and one gets an error.


Here is an example for ResNet152 using this [image](https://cdn.pixabay.com/photo/2015/05/07/11/02/guitar-756326_960_720.jpg). 

```                                                                                                                                             
Flux                                                                                                                                                                  
acoustic guitar: 0.9386926                                                                                                                                            
banjo: 0.019070234                                                                                                                                                    
electric guitar: 0.017291745                                                                                                                                          
stage: 0.0114803575                                                                                                                                                   
pick: 0.008877478        
                                                                                                                                             
PyTorch                                                                                                                                                               
acoustic guitar: 0.9386926                                                                                                                                            
banjo: 0.01907025                                                                                                                                                     
electric guitar: 0.017291661                                                                                                                                          
stage: 0.01148028                                                                                                                                                     
pick: 0.008877401         
```

**Comment by darsnack** [comment]
That's great! #111 should address the concerns you raised. Can you test your script with Metalhead's master branch once that is merged? If the validation checks out, then a PR to MetalheadWeights will be greatly appreciated!

**Comment by Alexander-Barth** [comment]
I would be glad to do that! 😀

For reference, the attached file contains the sizes of all trainable parameters of the Flux model generated by:

```julia
write("resnet50_sizes.txt",join(string.(size.(Flux.params(model))),"\n"))
```

If these sizes match, it will be quite likely that the `Metalhead.ResNet50` model is equivalent to the PyTorch model.

[resnet50_sizes.txt](https://github.com/FluxML/Metalhead.jl/files/8016782/resnet50_sizes.txt)
